### PR TITLE
feat: add target for building sage

### PR DESCRIPTION
**Pull request by Holmus**
When migrating to sage in a repository yesterday I found that quite a lot of time I would go back into the sage directory to regenerate my Makefile. Adding a target for doing this could be useful.

### CODE REVIEW

**Comment by Edholm** [comment]
What is the reason for addition? :thinking: 

Usually a `touch .sage/go.mod && make` should cover this use-case

**Comment by Holmus** [comment]
> What is the reason for the addition? thinking
> 
> Usually a `touch .sage/go.mod && make` should cover this use-case

I find using a target for this easier.

**Review by Edholm — COMMENTED** [review]
> > What is the reason for the addition? thinking
> > Usually a `touch .sage/go.mod && make` should cover this use-case
> 
> I find using a target for this easier.

I have nothing against it. Have you tried it in your repo? Does it work the way you expect it to?

I'll leave it to @odsod or @viktorvoltaire to approve though

**Comment by Holmus** [comment]
> > > What is the reason for the addition? thinking
> > > Usually a `touch .sage/go.mod && make` should cover this use-case
> > 
> > 
> > I find using a target for this easier.
> 
> I have nothing against it. Have you tried it in your repo? Does it work the way you expect it to?
> 
> I'll leave it to @odsod or @viktorvoltaire to approve though

I tried it in this repository. Have not tried it in any other repository.

File: Makefile
Comment (odsod) [inline Makefile:11]: I have a few thoughts here:

1) Building the sagefile binary is so quick now that we have no dependencies that I think we should make it a PHONY target and always rebuild it
2) I think that this target should simply be an alias for the $(sagefile) target, since they do the same thing

Comment (Holmus) [reply Makefile:11]: @viktorvoltaire could you chime in here?

Comment (viktorvoltaire) [reply Makefile:11]: I like how you are thinking @odsod just tested it now and building takes half a second :) lets go for it! @Holmus sure I can push a fix

Comment (viktorvoltaire) [reply Makefile:12]: Fixed! @Holmus @odsod @Edholm  what do think of the name `build-sage` ? another option could just be `make sage` :)

Comment (Holmus) [reply Makefile:12]: I kinda like `make sage`

Comment (Edholm) [reply Makefile:12]: :heavy_plus_sign: on `make sage`

Comment (viktorvoltaire) [reply Makefile]: sage it is!

**Comment by viktorvoltaire** [comment]
Even though I can approve this, I think its better if @Edholm and @odsod does it :rofl:

**Review by Edholm — APPROVED** [review]
LGTM!

**Review by odsod — APPROVED** [review]
Sweet

**Comment by odsod** [comment]
@viktorvoltaire Should we add an `update-sage` target while we're at it...? Breaking changes and convenience considered

**Comment by viktorvoltaire** [comment]
> Breaking changes and convenience considered

What do you purpose this target does?

**Comment by viktorvoltaire** [comment]
I will merge this, and we can look into the update command later

**Pull request merged**
